bucket_writer.go

```diff
@@ -0,0 +1,83 @@
+package bbolt
+
+import (
+	"bytes"
+	"encoding/binary"
+	"go.etcd.io/bbolt/internal/common"
+	"unsafe"
+)
+
+func EncodeBucketIntoBytes(bk *Bucket) []byte {
+	var buf []byte
+	bucketPages := fetchAllPagesInBucket(bk)
+	for _, bk := range bucketPages {
+		buf = append(buf, encodePage(bk)...)
+	}
+
+	return buf
+}
+
+func encodePage(page *common.Page) []byte {
```

Comment:
steps 

- encode pageHeader 
- encode fixed size part `leafPageElements` 
- encode variable length `keyValuePairs`

This impl is not efficient, and it does a lot of copying 

I traversed the repo looking for efficient way to write the pages, i could not find 

i found  a way to readPage from a file, not to write